hooks/FileDrop.ts

```diff
@@ -1,14 +1,20 @@
-import { markdownToHtml } from "../libs/markdown";
-import { Editor } from "tinymce";
+export const handleDrop = (): Promise<string | null> => {
+  return new Promise((resolve) => {
+    const handleFileDrop = async (e: DragEvent) => {
+      e.preventDefault();
+      const file = e.dataTransfer?.files[0];
+      if (file && file.name.endsWith('.md')) {
+        try {
+          const md = await file.text();
+          window.removeEventListener('drop', handleFileDrop);
```

Comment:
wont't this make fileDrop only works for first time since it remove event listener after uploading finished?
oh, is it because useEffect(xxx, []) will add event listener again after re-render?

Reply:
youre right!

updated